xbmc/playlists/PlayList.cpp

```diff
@@ -95,10 +95,6 @@ void CPlayList::Add(const CFileItemPtr &item, int iPosition, int iOrder)
   else
     item->m_iprogramCount = iOrder;
 
-  // videodb files are not supported by the filesystem as yet
-  if (item->IsVideoDb())
-    item->SetPath(item->GetVideoInfoTag()->m_strFileNameAndPath);
-
```

Comment:
You need to workaround the cases here, when VideoDataBaseFS fails and fall back to the item->SetPath way of doing it.